### PR TITLE
docs: add sample

### LINKED ISSUE #3: Add samples
Provide some complete mapper samples. Samples are a great way to get started for newer users or to check how specific features can be used.

### PR DESCRIPTION
**Pull request by CommonGuy**
closes #3

### CODE REVIEW

File: docs/prebuild.js [hunk 1]
```diff
@@ -79,8 +79,32 @@ async function buildAnalyzerRulesData() {
...
+
+  const sampleProject = '../samples/Riok.Mapperly.Sample';
+  const projectFilesToCopy = ['CarMapper.cs', 'Car.cs', 'CarDto.cs'];
+  const generatedMapperFile = join(sampleProject, 'obj/Debug/net7.0/generated/Riok.Mapperly/Riok.Mapperly.MapperGenerator/CarMapper.g.cs');
+
+  // clean target directory
+  await deleteFilesWithExtension(targetDir, '.cs');
+
+  // build sample project
+  await execPromise(`dotnet build ${sampleProject}`);
```
Comment (latonz) [inline docs/prebuild.js:94]: The project should always be built at this time (same precondition for the api docs).

File: samples/Riok.Mapperly.Sample/CarDto.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    public CarColorDto Color { get; set; }
+
+    public ProducerDto? Producer { get; set; }
+
+    public List<TireDto>? Tires { get; set; }
+}
+
+public enum CarColorDto
+{
+    Black,
```
Comment (latonz) [inline samples/Riok.Mapperly.Sample/CarDto.cs:18]: Maybe introduce an undefined value here and map this enum by name to showcase the enum mapping capabilities?

**Review by latonz — CHANGES_REQUESTED** [review]

File: docs/prebuild.js (hunk 1 above)
Comment (CommonGuy) [reply docs/prebuild.js]: Removed

File: samples/Riok.Mapperly.Sample/CarDto.cs (hunk 2 above)
Comment (CommonGuy) [reply samples/Riok.Mapperly.Sample/CarDto.cs]: Added different values to showcase this

**Review by latonz — APPROVED** [review]

**Pull request merged**